spec/manifolds/cli_spec.rb

```diff
@@ -3,86 +3,102 @@
+
+  before do
+    FakeFS do
+      # Set up any template files that need to exist
+      FileUtils.mkdir_p("#{File.dirname(__FILE__)}/../../lib/manifolds/templates")
+      File.write("#{File.dirname(__FILE__)}/../../lib/manifolds/templates/config_template.yml", "vectors:\nmetrics:")
+      File.write("#{File.dirname(__FILE__)}/../../lib/manifolds/templates/vector_template.yml", "attributes:")
+    end
+  end
 
   describe "#init" do
     subject(:cli) { described_class.new(logger: null_logger) }
 
-    before do
-      allow(FileUtils).to receive(:mkdir_p)
-      allow(File).to receive(:open)
-      cli.init(project_name)
-    end
+    context "when initializing a new project" do
+      before { cli.init(project_name) }
 
-    it "creates the projects directory" do
-      expect(FileUtils).to have_received(:mkdir_p).with("./#{project_name}/projects")
+      it "creates a project directory structure" do
+        expect(Dir.exist?("./#{project_name}")).to be true
+        expect(Dir.exist?("./#{project_name}/projects")).to be true
+        expect(Dir.exist?("./#{project_name}/vectors")).to be true
+      end
     end
   end
 
   describe "#add" do
-    let(:cli) { described_class.new(logger: null_logger) }
+    subject(:cli) { described_class.new(logger: null_logger) }
 
-    context "when within an umbrella project" do
+    context "when adding a project within an umbrella project" do
       before do
-        FileUtils.mkdir_p("#{project_name}/projects") # Simulate an umbrella project
+        FileUtils.mkdir_p("#{project_name}/projects")
         Dir.chdir(project_name)
         cli.add(sub_project_name)
       end
 
       after do
         Dir.chdir("..")
-        FileUtils.rm_rf(project_name)
       end
 
-      it "creates a tables directory within the project" do
+      it "creates the expected project structure" do
         expect(Dir.exist?("./projects/#{sub_project_name}/tables")).to be true
-      end
-
-      it "creates a routines directory within the project" do
         expect(Dir.exist?("./projects/#{sub_project_name}/routines")).to be true
       end
 
-      it "creates a manifold.yml file" do
-        expect(File.exist?("./projects/#{sub_project_name}/manifold.yml")).to be true
-      end
-
-      it "writes the manifold.yml file with dimensions" do
-        expect(File.read("./projects/#{sub_project_name}/manifold.yml")).to include("dimensions")
-      end
-
-      it "writes the manifold.yml file with metrics" do
-        config = File.read("./projects/#{sub_project_name}/manifold.yml")
-        expect(config).to include("metrics")
+      it "creates a manifold configuration file" do
+        config_file = "./projects/#{sub_project_name}/manifold.yml"
+        expect(File.exist?(config_file)).to be true
+        expect(File.read(config_file)).to include("vectors:")
+        expect(File.read(config_file)).to include("metrics:")
       end
     end
```

Comment:
You can build all this into a test with a single assertion. Testing that the file exists is redundant—if it doesn't exist the example will fail with an exception. On that subject, you should actually parse the file instead of looking for strings.

```ruby
it "creates a manifold configuration file" do
  cli.add(sub_project_name)
  config = YAML.safe_load_file("./projects/#{sub_project_name}/manifold.yml")
  expect(config).to include("vectors", "metrics")
end
```